### ISSUE TITLE
dart migrate: Aborting migration due to an exception

### ISSUE THREAD

**Issue by NaarGes** [dart-lang/sdk#45888]
Hi. I am trying to migrate my flutter app to null-safety with `dart migrate` command and I got the following error.
Dart SDK version: `2.12.3`
Operating System: `Windows  #10`
[code](https://gist.github.com/NaarGes/94689c2cd9daadfed5baeb339c407d8a)
 

```
Exception details:

UnimplementedError: LUB/GLB with unexpected types: InterfaceTypeImpl/DynamicTypeImpl in "buildArrangement ?? defaultArranger" on line 65 of "C:\Users\ASUS\Documents\Avicenna\nullsafe\me.mamania.flutterapp\lib\ui\shared\component\option_selector\radio_generator\radi
o_grid_gernerator.dart" at offset 2084 in C:\Users\ASUS\Documents\Avicenna\nullsafe\me.mamania.flutterapp\lib\ui\shared\component\option_selector\radio_generator\radio_grid_gernerator.dart (buildArrangement ?? defaultArranger)

#0      EdgeBuilder._unimplemented (package:nnbd_migration/src/edge_builder.dart:3253:5)
#1      EdgeBuilder._decorateUpperOrLowerBound (package:nnbd_migration/src/edge_builder.dart:2129:11)
#2      EdgeBuilder._decorateUpperOrLowerBound (package:nnbd_migration/src/edge_builder.dart:2172:40)
#3      EdgeBuilder.visitBinaryExpression (package:nnbd_migration/src/edge_builder.dart:495:26)
#4      BinaryExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:967:49)
#5      EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2207:24)
#6      EdgeBuilder.visitParenthesizedExpression (package:nnbd_migration/src/edge_builder.dart:1358:18)
#7      ParenthesizedExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:7964:15)
#8      EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2207:24)
#9      EdgeBuilder._checkExpressionNotNull (package:nnbd_migration/src/edge_builder.dart:1980:20)
#10     EdgeBuilder.visitFunctionExpressionInvocation (package:nnbd_migration/src/edge_builder.dart:962:32)
#11     FunctionExpressionInvocationImpl.accept (package:analyzer/src/dart/ast/ast.dart:5066:15)
#12     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2207:24)
#13     EdgeBuilder._handleAssignment (package:nnbd_migration/src/edge_builder.dart:2329:20)
#14     EdgeBuilder.visitReturnStatement (package:nnbd_migration/src/edge_builder.dart:1498:7)
#15     ReturnStatementImpl.accept (package:analyzer/src/dart/ast/ast.dart:8617:49)
#16     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2207:24)
#17     EdgeBuilder.visitNode (package:nnbd_migration/src/edge_builder.dart:1340:9)
#18     GeneralizingAstVisitor.visitStatement (package:analyzer/dart/ast/visitor.dart:517:39)
#19     GeneralizingAstVisitor.visitBlock (package:analyzer/dart/ast/visitor.dart:165:31)
#20     BlockImpl.accept (package:analyzer/src/dart/ast/ast.dart:1083:49)
#21     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2207:24)
#22     EdgeBuilder.visitNode (package:nnbd_migration/src/edge_builder.dart:1340:9)
#23     GeneralizingAstVisitor.visitFunctionBody (package:analyzer/dart/ast/visitor.dart:324:45)
#24     GeneralizingAstVisitor.visitBlockFunctionBody (package:analyzer/dart/ast/visitor.dart:168:55)
#25     BlockFunctionBodyImpl.accept (package:analyzer/src/dart/ast/ast.dart:1038:49)
#26     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2207:24)
#27     EdgeBuilder._handleExecutableDeclaration (package:nnbd_migration/src/edge_builder.dart:2488:7)
#28     EdgeBuilder.visitMethodDeclaration (package:nnbd_migration/src/edge_builder.dart:1243:5)
#29     MethodDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:7033:49)
#30     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2207:24)
#31     EdgeBuilder._dispatchList (package:nnbd_migration/src/edge_builder.dart:2229:7)
#32     EdgeBuilder.visitClassOrMixinOrExtensionDeclaration (package:nnbd_migration/src/edge_builder.dart:607:7)
#33     EdgeBuilder.visitClassDeclaration (package:nnbd_migration/src/edge_builder.dart:574:5)
#34     ClassDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:1536:49)
#35     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2207:24)
#36     EdgeBuilder.visitNode (package:nnbd_migration/src/edge_builder.dart:1340:9)
#37     GeneralizingAstVisitor.visitCompilationUnit (package:analyzer/dart/ast/visitor.dart:202:51)
#38     CompletenessTracker.visitCompilationUnit.<anonymous closure> (package:nnbd_migration/src/utilities/completeness_tracker.dart:52:24)
#39     PermissiveModeVisitor.reportExceptionsIfPermissive (package:nnbd_migration/src/utilities/permissive_mode.dart:26:24)
#40     CompletenessTracker.visitCompilationUnit (package:nnbd_migration/src/utilities/completeness_tracker.dart:43:5)
#41     CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2134:49)
#42     NullabilityMigrationImpl.processInput (package:nnbd_migration/src/nullability_migration_impl.dart:244:12)
#43     NonNullableFix.processUnit (package:nnbd_migration/src/front_end/non_nullable_fix.dart:164:15)
#44     _FixCodeProcessor.runLaterPhases.<anonymous closure> (package:nnbd_migration/migration_cli.dart:1154:19)
#45     _FixCodeProcessor.runLaterPhases.<anonymous closure> (package:nnbd_migration/migration_cli.dart:1152:28)
#46     _FixCodeProcessor.processResources (package:nnbd_migration/migration_cli.dart:1081:30)
<asynchronous suspension>
#47     _FixCodeProcessor.runLaterPhases (package:nnbd_migration/migration_cli.dart:1152:5)
<asynchronous suspension>
#48     MigrationCliRunner.run (package:nnbd_migration/migration_cli.dart:749:24)
<asynchronous suspension>
#49     MigrateCommand.run (package:nnbd_migration/migration_cli.dart:185:7)
<asynchronous suspension>
#50     CommandRunner.runCommand (package:args/command_runner.dart:196:13)
<asynchronous suspension>
#51     DartdevRunner.runCommand (package:dartdev/dartdev.dart:199:18)
<asynchronous suspension>
#52     runDartdev (package:dartdev/dartdev.dart:53:16)
<asynchronous suspension>
#53     main (file:///C:/b/s/w/ir/cache/builder/src/third_party/dart/pkg/dartdev/bin/dartdev.dart:11:3)
<asynchronous suspension>

```

**Comment by stereotype441** [comment]
This looks like a duplicate of #45574, which was fixed in 3186fa521ea323eb4e7fd8f05ce8f697b6c897b0 and is currently available in the dev channel (you need Dart version 2.14.0-6.0.dev or greater to have the fix).

If you don't want to try using a release from the dev channel, you should be able to work around the problem by supplying the `--ignore-exceptions` option when running the migration tool.

**Comment by stereotype441** [comment]
Duplicate of #45574
